pkg/services/baremetal/host/host.go

```diff
@@ -1181,13 +1189,43 @@ func verifyConnectionRefused(sshClient sshclient.Client, port int) bool {
 	return true
 }
 
-func (s *Service) actionEnsureProvisioned() actionResult {
+func (s *Service) actionEnsureProvisioned() (ar actionResult) {
 	sshClient := s.scope.SSHClientFactory.NewClient(sshclient.Input{
 		PrivateKey: sshclient.CredentialsFromSecret(s.scope.OSSSHSecret, s.scope.HetznerBareMetalHost.Spec.Status.SSHSpec.SecretRef).PrivateKey,
 		Port:       s.scope.HetznerBareMetalHost.Spec.Status.SSHSpec.PortAfterCloudInit,
 		IP:         s.scope.HetznerBareMetalHost.Spec.Status.GetIPAddress(),
 	})
 
+	defer func() {
+		// Create an Event which contains the content of /var/log/cloud-init-output.log
+
+		if _, ok := ar.(actionContinue); ok {
+			// don't create an event
+			return
+		}
+		out := sshClient.GetCloudInitOutput()
+		if out.Err != nil || out.StdErr != "" {
+			record.Warnf(s.scope.HetznerBareMetalHost, "GetCloudInitOutputFailed",
+				fmt.Sprintf("GetCloudInitOutput failed: stdout %q, stderr %q, err %q",
+					out.StdOut, out.StdErr, out.Err.Error()))
+			return
+		}
+		_, ok := ar.(actionComplete)
+		if ok {
```

Comment:
you don't need this if else statement.. You can just make a simple one where you define errString and then you just append the stdout of cloud init to the error string (which might be empty)